### PR TITLE
[stable25] Check for folder existence in FolderController

### PR DESCRIPTION
**Pull request by Antreesy**
- [x] :warning: This backport had conflicts that were resolved with the 'ours' merge strategy and is likely incomplete

Manual backport of #2375
Please check for changes

### CODE REVIEW

**Comment by Antreesy** [comment]
Smoke tested, still able to create, rename, delete folder, and add user groups
@juliushaertl please also check manually, there are some changes i've resolved myself

**Review by juliushaertl — APPROVED** [review]

File: lib/Controller/FolderController.php [hunk 1]
```diff
@@ -84,10 +84,28 @@ public function getFolders(): DataResponse {
...
+		}
+
+		$storageId = $this->getRootFolderStorageId();
+		return new DataResponse($this->manager->getFolder($id, $storageId));
+	}
+
+	private function checkFolderExists(int $id): ?DataResponse {
+		$storageId = $this->getRootFolderStorageId();
+		if ($storageId === null) {
+			return new DataResponse([], Http::STATUS_NOT_FOUND);
```
Comment (juliushaertl) [inline lib/Controller/FolderController.php:99]: The class is not found (see the failing static analysis check) as the `use OCP\AppFramework\Http;` declaration is missing in the file (L28)

**Comment by Antreesy** [comment]
Now it complains on PHP dependencies in runner, I guess?
```
PHP module GD not installed.
Please ask your server administrator to install the module.
```

**Comment by Antreesy** [comment]
Rebased and squashed commits

**Pull request merged**